packages/vetrina/src/Vetrina/Purchase/NewPurchase.purs

```diff
@@ -40,14 +40,15 @@ type State =
   }
 
 type Props =
-  { accountStatus :: AccountStatus
-  , products :: Array Product
-  , mkPurchaseWithNewAccount :: NewAccountForm -> Effect Unit
+  { accountStatus                 :: AccountStatus
+  , products                      :: Array Product
+  , errorMessage                  :: JSX
```

Comment:
Let's change this to `String` and let `NewPurchase` decide what `JSX` to render.

Reply:
But this way `NewPurchase` needs to know what kind of order failure was there correct? At least if I understood what you mean by let `NewPurchase` decide what `JSX` to render.

Reply:
If we are always showing the error above the input field, it's enough to know what the message is, right?

So the thing in `Vetrina` becomes:
```purescript
orderErrorMessage :: OrderFailure -> String
orderErrorMessage failure =
  case failure of
    AuthenticationError -> "Kombinationen av e-postadress och lösenord finns inte"
    _ -> DOM.text "Något gick fel. Vänligen försök om en stund igen."
```